wpeverest/everest-forms
# Feature - Mollie payment integration

### All Submissions:






### Changes proposed in this Pull Request:

Previously, in Everest Forms there was no option for Mollie payment gateway integration. This PR allows the to integrate the normal Mollie payment and the recurring payment subscription using the Mollie payment.

Closes # .

### How to test the changes in this Pull Request:

1. Checkout to this branch
2. Go to the EverestForms Dashboard>Features>Mollie
3. Enable the **Mollie** and click on the settings button after activating the feature.
4. The settings will navigate to the global settings>Payment Section.
5. Now enter the **Test API key** if you are using the **Test Mode**.
6. In order to use the **Test Mode** hit the **Enable Test Mode** toggle button.
8. You can retrieve the [Test API key](https://docs.mollie.com/reference/testing) and [create the account](https://help.mollie.com/hc/en-gb/articles/210709969-How-do-I-create-an-account).
9. Go to Everest Forms>Add new.
10. Drag and drop the payment fields.
11. Go to Form settings>Payments>Mollie. 
12. Enable the Mollie for simple payment.
13. Now preview the check whether it is working properly or not.

**Enabling the Recurring payment**
1. All steps upto enabling **Mollie** payment is same.
2. Now under the **Subscription** you can find the **Enable recurring subscription payments on Mollie**
3. Enable the toggle to use recurring payment.
4. Enter the **Subscription Description**, **Recurring Duration**, **Time period of the recurring payment** respectively.
5. Now Map the **Customer's Email**, **Customer's First Name**, **Customer's Last Name**. **Note:** These three fields are the required fields.
6. Now Perform the entry and check whether the customer and subscription is created or not from the **Mollie Dashboard**.

[Everest Forms Pro PR Dependency](https://github.com/wpeverest/everest-forms-pro/pull/909)
**Note** Regarding the backward compatibility if **other payment gateways are enabled then Mollie won't be working by default.**
**Note** You need to activate the **Cards** From **SiteName>Organization Settings>Products>Payments**. I have attached image for reference.
[First Step](https://i.imgur.com/nHkFDbg.png)
[Second Step](https://i.imgur.com/44DEJEW.png)
[Third Step](https://i.imgur.com/y0YQCbC.png)

### Types of changes:




### Other information:




### Changelog entry

> Feature - Mollie payment integration
